### PR TITLE
Fix for Newline with EncapsedStringsToSprintfRector

### PR DESCRIPTION
**Pull request by brucealdridge**
Fix for #5032

### CODE REVIEW

File: rules/coding-style/src/Rector/Encapsed/EncapsedStringsToSprintfRector.php [hunk 1]
```diff
@@ -123,8 +123,16 @@ private function collectExpr(Expr $expr): void
     private function createSprintfFuncCallOrConcat(string $string, array $argumentVariables): Node
     {
         // special case for variable with PHP_EOL
-        if (count($argumentVariables) === 2 && $argumentVariables[1] instanceof ConstFetch && (string) $argumentVariables[1]->name === 'PHP_EOL') {
-            return new Concat($argumentVariables[0], $argumentVariables[1]);
+        if ($string === '%s%s' && count($argumentVariables) === 2 &&
+            count(
+                array_filter(
+                    $argumentVariables,
+                    fn($argumentVar): bool => $argumentVar instanceof ConstFetch && $argumentVar->name.'' === 'PHP_EOL'
```
Comment (samsonasik) [inline rules/coding-style/src/Rector/Encapsed/EncapsedStringsToSprintfRector.php:130]: argument->name . '' should can be (string) argument->name

Comment (brucealdridge) [reply rules/coding-style/src/Rector/Encapsed/EncapsedStringsToSprintfRector.php:130]: @samsonasik thanks, guessing you're also building against >7.3 as it doesn't like the return type on the closure

**Comment by samsonasik** [comment]
@brucealdridge I cherry-picked your commit at #5072

**Comment by brucealdridge** [comment]
@samsonasik thank you! Haven't yet had the time to rewrite!

**Pull request closed without merging**